internal/pkg/extensions/kernel_modules.go

```diff
@@ -105,14 +105,11 @@ func GenerateKernelModuleDependencyTreeExtension(extensionsPathWithKernelModules
 		return nil, fmt.Errorf("error running depmod: %w", err)
 	}
 
-	// we want this temp directory to be present until the extension is compressed later on, so not removing it here
-	kernelModulesDependencyTreeStagingDir, err := os.MkdirTemp("", "module-deps")
-	if err != nil {
-		return nil, err
-	}
+	// we want the extension to survive this function, so not storing in a temporary directory
```

Comment:
I fixed this to use an existing extension tree path instead of temp dir, otherwise it won't be cleaned up properly (leaking temporary directory)